metaconfig-core/shared/src/main/scala-3/metaconfig/generic/package.scala

```diff
@@ -0,0 +1,334 @@
+package metaconfig.generic
```

Comment:
This whole file 👏 👏 👏

Reply:
It was fun, actually :D Some things clicked after several hours of bashing my head against the only real source of ["documentation"](https://github.com/lampepfl/dotty/blob/master/library/src/scala/quoted/Quotes.scala)

I was pretty stuck until I "discovered" a WIP macro tutorial: https://docs.scala-lang.org/scala3/guides/macros/quotes.html which actually showed several examples of exactly the things I needed.